api/src/services/common_grants/transformation.py

```diff
@@ -0,0 +1,461 @@
+from src.constants.lookup_constants import OpportunityStatus
+from src.db.models.opportunity_models import Opportunity, OpportunitySummary
+from src.validation.validation_constants import ValidationErrorType
+
+logger = logging.getLogger(__name__)
+
+
+def transform_status_to_cg(v1_status: OpportunityStatus) -> OppStatusOptions:
+    """
+    Transform v1 enum to CG enum.
+
+    Args:
+        OpportunityStatus: The v1 enum value
+
+    Returns:
+        OppStatusOptions: The CG enum value
+    """
+
+    STATUS_TO_CG_MAP = {
+        OpportunityStatus.FORECASTED: OppStatusOptions.FORECASTED,
+        OpportunityStatus.POSTED: OppStatusOptions.OPEN,
+        OpportunityStatus.CLOSED: OppStatusOptions.CLOSED,
+        OpportunityStatus.ARCHIVED: OppStatusOptions.CUSTOM,
+    }
+
+    cg_status = STATUS_TO_CG_MAP.get(v1_status, None)
+    if not cg_status:
+        logger.error(
+            f"Transform failed for field `status`: unexpected OpportunityStatus value: {v1_status}"
+        )
+        cg_status = OppStatusOptions.FORECASTED
+
+    return cg_status
+
+
+def transform_status_from_cg(cg_status: OppStatusOptions) -> str:
+    """
+    Transform CG enum value to v1 enum value.
+
+    Args:
+        OppStatusOptions: The CG enum value
+
+    Returns:
+        v1_status: The v1 enum value
+    """
+
+    STATUS_FROM_CG_MAP = {
+        OppStatusOptions.FORECASTED: OpportunityStatus.FORECASTED,
+        OppStatusOptions.OPEN: OpportunityStatus.POSTED,
+        OppStatusOptions.CLOSED: OpportunityStatus.CLOSED,
+        OppStatusOptions.CUSTOM: OpportunityStatus.ARCHIVED,
+    }
+
+    v1_status = STATUS_FROM_CG_MAP.get(cg_status, None)
+    if not v1_status:
+        logger.error(
+            f"Transform failed for field `status`: unexpected OpportunityStatusOptions value: {cg_status}"
+        )
+        v1_status = OpportunityStatus.FORECASTED
+
+    return v1_status
+
+
+def transform_sorting_from_cg(cg_sort_by: OppSortBy) -> str:
+    """
+    Transform CG enum value to v1 enum value.
+
+    Args:
+        OppSortBy: The CG enum value
+
+    Returns:
+        sort_by: The v1 enum value
+    """
+
+    SORT_FIELD_MAPPING = {
+        OppSortBy.LAST_MODIFIED_AT: "updated_at",
+        OppSortBy.CREATED_AT: "created_at",
+        OppSortBy.TITLE: "opportunity_title",
+        OppSortBy.STATUS: "opportunity_status",
+        OppSortBy.CLOSE_DATE: "close_date",
+        OppSortBy.MAX_AWARD_AMOUNT: "award_ceiling",
+        OppSortBy.MIN_AWARD_AMOUNT: "award_floor",
+        OppSortBy.TOTAL_FUNDING_AVAILABLE: "estimated_total_program_funding",
+    }
+
+    v1_sort_by = SORT_FIELD_MAPPING.get(cg_sort_by, None)
+
+    if not v1_sort_by:
+        logger.error(
+            f"Transform failed for field `sort_by`: unexpected OppSortBy value: {cg_sort_by}"
+        )
+        v1_sort_by = OppSortBy.LAST_MODIFIED_AT
+
+    return v1_sort_by
+
+
+def _get_opportunity_summary(opportunity: Opportunity) -> OpportunitySummary | None:
```

Comment:
The opportunity model has a property called `summary` that is this utility function, could do `opportunity.summary` and get the same result